src/web/app/pages-instructor/instructor-session-edit-page/instructor-session-edit-page.component.ts

```diff
@@ -780,6 +785,41 @@ export class InstructorSessionEditPageComponent extends InstructorSessionBasePag
     // TODO scroll down to the question specific section in the help page
   }
 
+  /**
+   * Gets all instructors of a course which can be previewed as.
+   */
+  getAllInstructorsCanBePreviewedAs(): void {
+    const paramMap: { [key: string]: string } = {
+      courseid: this.courseId,
+      intent: Intent.FULL_DETAIL,
+    };
+    this.httpRequestService.get('/instructors', paramMap)
+        .subscribe((instructors: Instructors) => {
+          this.instructorsCanBePreviewedAs = instructors.instructors;
+
+          // TODO use privilege API to filter instructors who has INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS
+          // in the feedback session
+
+          // sort the instructor list based on name
+          this.instructorsCanBePreviewedAs.sort((a: Instructor, b: Instructor): number => {
+            return a.name.localeCompare(b.name);
+          });
+
+          // select the first instructor
+          if (this.instructorsCanBePreviewedAs.length >= 1) {
+            this.emailOfInstructorToPreview = this.instructorsCanBePreviewedAs[0].email;
+          }
+        }, (resp: ErrorMessageOutput) => { this.statusMessageService.showErrorMessage(resp.error.message); });
+  }
+
+  /**
+   * Previews the submission of the feedback session as an instructor.
+   */
+  previewAsInstructor(): void {
+    window.open(`${environment.frontendUrl}/web/instructor/sessions/submission`
```

Comment:
Do you really need `environment.frontendUrl`?

Reply:
We need open the URL in a new tab. We might need to abstract the method in `NavigationService` in due course.